### PR TITLE
Adding sparse_dot_mkl

### PR DESCRIPTION
**Pull request by asistradition**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by asistradition** [comment]
@conda-forge/staged-recipes Following up on this, is there anything else that I need to do?

**Comment by asistradition** [comment]
@conda-forge/help-python This is ready for review, thanks

**Comment by asistradition** [comment]
@conda-forge-admin please ping team

This is ready for review when you get a chance - thanks!

File: recipes/sparse_dot_mkl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - python -c "import sparse_dot_mkl.tests.test_mkl; sparse_dot_mkl.tests.test_mkl.run()"
+
+about:
+  home: "https://github.com/flatironinstitute/sparse_dot"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "Intel MKL wrapper for sparse matrix multiplication"
+  doc_url: 
+  dev_url:
```
Comment (beckermr) [inline recipes/sparse_dot_mkl/meta.yaml:44]: Please remove these empty keys

File: recipes/sparse_dot_mkl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - numpy >=1.16
+    - python
+    - scipy >=1.0
+    - mkl >=v2020.0
```
Comment (beckermr) [inline recipes/sparse_dot_mkl/meta.yaml:25]: mkl version numbers do not have a `v` in the front on conda-forge

Comment (asistradition) [reply recipes/sparse_dot_mkl/meta.yaml]: Resolved

File: recipes/sparse_dot_mkl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 352341c5cf739f72f49ba3df972f5fe99b08b7d2dfebe14367cfa857d0f4e9e0
+
+build:
+  skip: true  # [win]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (beckermr) [inline recipes/sparse_dot_mkl/meta.yaml:15]: Can this recipe be noarch python?

Comment (asistradition) [reply recipes/sparse_dot_mkl/meta.yaml:15]: Yes; it specifically is for noarch. The conda-forge MKL dependency is broken for windows though. As long as I can leave a win skip or just override the build I redo the recipe as noarch.

Comment (beckermr) [reply recipes/sparse_dot_mkl/meta.yaml:16]: noarch recipes cannot use selectors. However, you do not need to skip windows. If the solver cannot find mkl for windows, it simply won't install the package. So remove the skip statement and add `noarch: python`

**Comment by beckermr** [comment]
A selector is the `# [win]`. You can and should keep the version requirements.

File: recipes/sparse_dot_mkl/meta.yaml (hunk 3 above)
Comment (asistradition) [reply recipes/sparse_dot_mkl/meta.yaml:15]: I am perfectly happy to merge the current `noarch` recipe that fails windows build check.

**Pull request merged**